src/test/java/org/apache/commons/text/StringSubstitutorWithInterpolatorStringLookupTest.java

```diff
@@ -68,6 +68,51 @@ public void testDefaultValueForMissingKeyInResourceBundle() {
         assertEquals("defaultValue", stringSubstitutor.replace("${keyWithMissingKey}"));
     }
 
+    @Test
+    public void testDnsLookup() throws UnknownHostException {
+        final StringSubstitutor strSubst = StringSubstitutor.createInterpolator();
+        Assertions.assertEquals(InetAddress.getByName("apache.org").getHostAddress(),
+            strSubst.replace("${dns:apache.org}"));
```

Comment:
All of the apache.org tests are also brittle because apache.org has two addresses. It can happen that you receive non equal anwers.